pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -996,6 +996,18 @@ type PrometheusSpec struct {
 	// Maximum number of bytes used by the Prometheus data.
 	RetentionSize ByteSize `json:"retentionSize,omitempty"`
 
+	// ShardRetentionPolicy defines the retention policy for the Prometheus shards.
+	// (Alpha) Using this field requires the 'PrometheusShardRetentionPolicy' feature gate to be enabled.
+	//
+	// The final goals for this feature can be seen at https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202310-shard-autoscaling.md#graceful-scale-down-of-prometheus-servers,
+	// however, the feature is not yet fully implemented and has known limitations:
+	// * Scale up after scale down will not re-use the previously marked-for-deletion shards.
+	// * Target redistribution doesn't happen on scale up/down.
+	// * Retention duration is not settable, for now shards are retained forever.
```

Comment:
I don't know if I follow this.
Are we retaining data forever? What does it means?

Reply:
I read it as: "if you choose policy=retain, the shards are never removed"